### PR TITLE
[tempfile] Add RBS for "tempfile"

### PR DESCRIPTION
**Pull request by tlvu2697**
Add RBS signatures for the "tempfile" package.

### CODE REVIEW

File: stdlib/tempfile/0/tempfile.rbs [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+  public
+
+  # Closes the file. If `unlink_now` is true, then the file will be unlinked
+  # (deleted) after closing. Of course, you can choose to later call #unlink if
+  # you do not unlink it now.
+  #
+  # If you don't explicitly unlink the temporary file, the removal will be delayed
+  # until the object is finalized.
+  #
+  def close: (?bool unlink_now) -> bool?
```
Comment (soutaro) [inline stdlib/tempfile/0/tempfile.rbs:148]: > __[Goodcheck]__ Prefer `boolish` over `bool` for method arguments and block return values ([view](https://sider.review/gh/repos/174796313/pulls/767#issue-5983947))
>
> <details>
>
> See the doc below:
> https://github.com/ruby/rbs/blob/78d04a2db0f1c4925d2b13c2939868edf9465d6c/docs/syntax.md#bool-or-boolish
>
> | Rule |
> | ---- |
> | `rbs.prefer_boolish` |
>
> You can [close this issue](https://sider.review/gh/repos/174796313/pulls/767#issue-5983947) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

Could you make the type of the parameter `boolish`?

Comment (tlvu2697) [reply stdlib/tempfile/0/tempfile.rbs:148]: I tried but then got into this trouble
```
  def test_close
    assert_send_type "(?boolish unlink_now) -> bool?",
                     Tempfile.new('README.md'), :close

    assert_send_type "(?boolish unlink_now) -> bool?",
                     Tempfile.new('README.md'), :close, true

    assert_send_type "(?boolish unlink_now) -> bool?",
                     Tempfile.new('README.md'), :close, false
  end

```

```
Started
....E
Error: test_close(TempfileTest): RuntimeError: Unknown name for expand_alias: boolish
/rbs/lib/rbs/definition_builder.rb:784:in `expand_alias'
/rbs/lib/rbs/test/type_check.rb:319:in `value'
/rbs/lib/rbs/test/type_check.rb:79:in `block in args'
/rbs/lib/rbs/test/type_check.rb:178:in `block in zip_args'
<internal:kernel>:148:in `yield_self'
/rbs/lib/rbs/test/type_check.rb:176:in `zip_args'
/rbs/lib/rbs/test/type_check.rb:78:in `args'
/rbs/lib/rbs/test/type_check.rb:50:in `method_call'
/rbs/test/stdlib/test_helper.rb:265:in `assert_send_type'
/rbs/test/stdlib/tempfile/Tempfile_test.rb:54:in `test_close'
     51:     assert_send_type "(?boolish unlink_now) -> bool?",
     52:                      Tempfile.new('README.md'), :close
     53:
  => 54:     assert_send_type "(?boolish unlink_now) -> bool?",
     55:                      Tempfile.new('README.md'), :close, true
     56:
     57:     assert_send_type "(?boolish unlink_now) -> bool?",
=======================================================================================
........
Finished in 0.008122501 seconds.
---------------------------------------------------------------------------------------
13 tests, 34 assertions, 0 failures, 1 errors, 0 pendings, 0 omissions, 0 notifications
92.3077% passed
---------------------------------------------------------------------------------------
```
I'm not quite sure about the error here, please help me with this :pray:

Comment (soutaro) [reply stdlib/tempfile/0/tempfile.rbs:148]: Yeah, you can write `::boolish`. But, I recommend writing the specific types of the arguments and return value.

```
  def test_close
    assert_send_type "() -> void",
                     Tempfile.new('README.md'), :close

    assert_send_type "(true) -> void",
                     Tempfile.new('README.md'), :close, true

    assert_send_type "(String) -> void",
                     Tempfile.new('README.md'), :close, "true"

    assert_send_type "(false) -> void",
                     Tempfile.new('README.md'), :close, false
  end
```

I think `(?boolish) -> void` would be the correct type here.

Comment (tlvu2697) [reply stdlib/tempfile/0/tempfile.rbs]: I've updated the code. Thank you so much for your support :tada:

**Pull request merged**